app/views.py

```diff
@@ -1,28 +1,370 @@
 '''This is where code for api resources will go'''
-
+import json
 from flask_restful import Resource
+from .models import Menu, Meal, User, Admin, Order, Database
+from flask import request
+from . import  app_db
+# app_db = Database()
 
 
 class UserRegistrationResource(Resource):
     '''Manage user registration when method is POST'''
-    pass
+    def post(self):
+        '''handle the POST request to register users'''
+        try:
+            post_data = json.loads(request.data)
+            username = post_data.get('username')
+            password = post_data.get('password')
+            email = post_data.get('email')
+            admin = post_data.get('admin', '')
+            if admin:
+                # register user using Admin model
+                admin = Admin(
+                    username=username, password=password, email=email,
+                    admin=admin)
+                try:
+                    app_db.add(item=admin)
+                except ItemAlreadyExists as e:
+                    return {
+                        'message': 'User already exists',
+                        'error': str(e)
+                    }, 202
+                return {
+                    'message': 'Admin registration succesful, proceed to login'
+                }, 201
+
+            else:
+                # register normal user
+                user = User(username=username, password=password, email=email)
+                try:
+                    app_db.add(item=user)
+                except Exception as e:
+                    return {
+                        'message': 'User already exists',
+                        'error': str(e)
+                    }, 202
+                return {
+                    'message': 'User registration succesful, proceed to login'
+                    }, 201
+
+        except Exception as e:
+            return {
+                'message': 'Encountered an error during registration',
+                'Error': str(e)
+            }, 400
 
 
 class LoginResource(Resource):
     '''Manage user log in'''
-    pass
+    def post(self):
+        try:
+            post_data = json.loads(request.data)
```

Comment:
For efficiency, just use `request.get_json()`

Reply:
changed all json.loads() to request.get_json()